pkg/app/api/api/web_api.go

```diff
@@ -334,7 +334,10 @@ func (a *WebAPI) getPiped(ctx context.Context, pipedID string) (*model.Piped, er
 // It gives back error unless the piped belongs to the project.
 func (a *WebAPI) validatePipedBelongsToProject(ctx context.Context, pipedID, projectID string) error {
 	pid, err := a.pipedProjectCache.Get(pipedID)
-	if err == nil && pid == projectID {
+	if err == nil {
+		if pid != projectID {
+			return status.Error(codes.PermissionDenied, "Requested piped doesn't belong to the project you logged in")
+		}
```

Comment:
Btw, It would be nice if we have some tests for those validation functions.

Reply:
Right, it would be testable.